### PR TITLE
add package automation scripts and templates

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: automation/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+```
+podman build --tag tfm-pulpcore-builder .
+```
+
+### Run
+
+1. create an empty directory to host the resulting packages
+2. pass the directory as a volume for `/app/packaging` to the container:
+```
+podman run --rm --volume $(pwd)/packaging/:/app/packaging/:Z  --env-file packaging.env tfm-pulpcore-builder
```
Comment (ekohl) [inline automation/README.md:37]: I get SELinux denials when I share volumes. Is there some boolean I'm missing?

File: automation/_generate_deps.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+PULPCORE_PACKAGES="pulpcore==3.6.2 pulp-file==1.2.0 pulp-container==2.0.0 pulp-deb==2.6.1 pulp-rpm==3.6.1 pulp-certguard==1.0.2 galaxy-importer==0.2.8 pulp-ansible==0.2.0"
```
Comment (ekohl) [inline automation/_generate_deps.sh]: Is it time for a requirements.txt file?

File: automation/README.md (hunk 1 above)
Comment (evgeni) [reply automation/README.md:37]: Can you post them? The above works on my F32 with Selinux in enforcing when `$(pwd)/packaging/` is a newly created dir in my home

File: automation/_generate_deps.sh (hunk 2 above)
Comment (evgeni) [reply automation/_generate_deps.sh]: modern shit, but yeah I guess

File: automation/README.md (hunk 1 above)
Comment (ekohl) [reply automation/README.md:37]: Ah, I didn't see the `:Z`. Looks like that changes the file context:
```console
$ mkdir packaging
$ ls -lZd packaging/
drwxrwxr-x. 2 ekohl ekohl unconfined_u:object_r:user_home_t:s0 4096 Sep  4 10:55 packaging/
$ podman run --rm --volume $(pwd)/packaging/:/app/packaging/:Z -it centos:7
[root@b240c45dee80 /]# exit
$ ls -lZd packaging/
drwxrwxr-x. 2 ekohl ekohl system_u:object_r:container_file_t:s0:c123,c1003 4096 Sep  4 10:55 packaging/
```

File: automation/_generate_deps.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  git config --local user.email "${EMAIL:-root@localhost}"
+  git config --local user.name "${NAME:-root}"
+  export RPM_PACKAGER="${NAME}"
+
+  while read line; do
+    pkg=${line%==*}
+    version=${line#*==}
+    REWRITE_ON_SAME_VERSION=false $FOREMAN_PACKAGING/add_pypi_package.sh ${pkg} ${version} $PULPCORE_TAG '/' $PULPCORE_PACKAGING/pyp2rpm/pulpcore.spec
+    sleep 5
```
Comment (ekohl) [inline automation/_generate_deps.sh:38]: Is this to prevent pypi from ratelimiting you? Can use a comment :)

File: pyp2rpm/pulpcore.spec [hunk 4]
```diff
@@ -0,0 +1,105 @@
+{{ data.credit_line }}
```
Comment (ekohl) [inline pyp2rpm/pulpcore.spec:1]: Should this use a `.j2` extension to indicate it's a template? Is that a common convention?

File: automation/README.md (hunk 1 above)
Comment (evgeni) [reply automation/README.md:37]: Yes, the :Z enforces it to be the right context for running the container.

File: pyp2rpm/pulpcore.spec (hunk 4 above)
Comment (evgeni) [reply pyp2rpm/pulpcore.spec:1]: We had that discussion when we added those templates in `foreman-packaging` initially: https://github.com/theforeman/foreman-packaging/pull/4318#discussion_r346842455

TL;DR: pyp2rpm would break :(

File: automation/README.md (hunk 1 above)
Comment (ekohl) [reply automation/README.md:37]: I learned something new today :)

File: automation/_generate_deps.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
```
Comment (ekohl) [inline automation/_generate_deps.sh:1]:
```suggestion
#!/bin/bash -e
```

File: automation/README.md (hunk 1 above)
Comment (evgeni) [reply automation/README.md:37]: The best part of it: on systems without SELinux passing :Z (or :z) to --volume breaks Docker/podman, so you need to do funny hacks if you want scripts to work on both worlds:

https://github.com/Katello/katello-client-bootstrap/blob/master/Makefile#L5

Comment (ehelms) [reply automation/README.md:37]: Would a little wrapper script that creates a default directory for me and just runs this command be asking too much?

Comment (evgeni) [reply automation/README.md:37]: Absolutely!

File: automation/_generate_deps.sh (hunk 5 above)
Comment (evgeni) [reply automation/_generate_deps.sh:1]: I prefer `set -e` instead, done

File: automation/update_packaging.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+mkdir -p ${RESULT_DIR}
+
+if [[ -f ${ENV_FILE} ]]; then
+  ENV_FILE_ARG="--env-file ${ENV_FILE}"
+else
+  ENV_FILE_ARG=""
+fi
+
+podman run --rm --volume ${RESULT_DIR}:/app/packaging/:Z  ${ENV_FILE_ARG} tfm-pulpcore-builder
```
Comment (ekohl) [inline automation/update_packaging.sh]: Now that there's a script, do you want to apply the `:Z` conditional based on SELinux as well?

File: automation/README.md (hunk 1 above)
Comment (ehelms) [inline automation/README.md:37]: Ought this mention the script now first?

**Review by ehelms — APPROVED** [review]
